### PR TITLE
301 Moved to Javascript Folder

### PR DESCRIPTION
**Pull request by johncokos**

### CODE REVIEW

**Comment by brookr** [comment]
@johncokos Is the script to download the challenges still delivering them to the correct path?

**Comment by brookr** [comment]
Also has conflicts

**Comment by brookr** [comment]
@johncokos Resolve merge conflicts, please, and complete testing.

**Comment by johncokos** [comment]
Conflicts cleared, code updated. Let's go!

**Comment by brookr** [comment]
I don't like naming the folders based on the course codes, makes it look more like school work, rather than a portfolio building repo. Can we combine the js challenges into a single folder?

File: README.md [hunk 1]
```diff
@@ -6,10 +6,19 @@ See [setup instructions](https://codefellows.github.io/setup-guide/code-301/3-co
 
 ### 301 Code Challenges
 
-Under the `data-structures-and-algorithms` repository, at the top level is a folder called `code-challenges`
+Under the `javascript` repository, at the top level is a folder called `code-challenges`
```
Comment (brookr) [inline README.md:9]: wrong wording?

File: javascript/301-code-challenges/bin/get-challenge.sh [hunk 2]
```diff
@@ -2,4 +2,4 @@
 
 echo "Getting Code Challenge $1"
 
-`which curl` -s https://codefellows.github.io/code-301-guide/curriculum/class-$1/challenges/challenges-$1.test.js > ./code-challenges/challenges-$1.test.js
+`which curl` -s https://codefellows.github.io/code-301-guide/curriculum/class-$1/challenges/challenges-$1.test.js > ./301-code-challenges/challenges-$1.test.js
```
Comment (brookr) [inline javascript/301-code-challenges/bin/get-challenge.sh:5]: Remove `301`

File: javascript/package.json [hunk 3]
```diff
@@ -4,17 +4,17 @@
   "description": "Data Structures, Algorithms, Code Challenges",
   "main": "index.js",
   "scripts": {
-    "get-challenge": "`which bash` ./code-challenges/bin/get-challenge.sh",
+    "get-challenge": "`which bash` ./301-code-challenges/bin/get-challenge.sh",
```
Comment (brookr) [inline javascript/package.json:7]: remove `301`

**Review by brookr — CHANGES_REQUESTED** [review]
Thanks, @johncokos !

**Comment by brookr** [comment]
Ready to merge in the week of May 17th.

**Pull request merged**